### ISSUE TITLE
Docker creates ingress network folder in /var/lib/docker even when using -g

### ISSUE THREAD

**Issue by lsapan** [docker/docker#31610]
**Description**

We are specifying a different docker root with `-g`, i.e. `-g /data/docker/`. It works, however docker is still recreating the `/var/lib/docker` folder and creates: `/var/lib/docker/network/files/ingress_sbox`.

Inside that folder are three files:
```
root@xxx:/var/lib/docker/network/files/ingress_sbox# ls -l
total 12
-rw-r--r-- 1 root root 150 Mar  7 13:18 hosts
-rw-r--r-- 1 root root  58 Mar  7 13:18 resolv.conf
-rw-r--r-- 1 root root  71 Mar  7 13:18 resolv.conf.hash
```

**Steps to reproduce the issue:**
1. Create a swarm
2. Create a service

**Describe the results you received:**
Docker is creating the `/var/lib/docker` folder.

**Describe the results you expected:**
Docker shouldn't create the `/var/lib/docker` folder.

**Output of `docker version`:**

```
Client:
 Version:      17.03.0-ce
 API version:  1.26
 Go version:   go1.7.5
 Git commit:   3a232c8
 Built:        Tue Feb 28 08:01:32 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.03.0-ce
 API version:  1.26 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   3a232c8
 Built:        Tue Feb 28 08:01:32 2017
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 20
 Running: 6
 Paused: 0
 Stopped: 14
Images: 20
Server Version: 17.03.0-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
Swarm: active
 NodeID: 9xet8ik8dju4y8786a2b3aemd
 Is Manager: true
 ClusterID: c935qq94yavddxpuc6292zbxu
 Managers: 3
 Nodes: 5
 Orchestration:
  Task History Retention Limit: 5
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
 Node Address: 172.31.70.70
 Manager Addresses:
  172.31.70.70:2377
  172.31.71.70:2377
  172.31.72.70:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 977c511eda0925a723debdc94d09459af49d082a
runc version: a01dafd48bc1c7cc12bdb01206f9fea7dd6feb70
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-64-generic
Operating System: Ubuntu 16.04.2 LTS
OSType: linux
Architecture: x86_64
CPUs: 1
Total Memory: 1.952 GiB
Name: ip-172-31-70-70
ID: FMBW:VTZX:TIKD:2IDU:P23J:LAOZ:5PAX:ZUOT:WUIY:NTES:LX5W:VR76
Docker Root Dir: /data/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: No swap limit support
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Swarm is hosted in AWS, we are specifying a second EBS volume (which is encrypted) with `-g`.

**Comment by cpuguy83** [comment]
ping @aboch

**Comment by imjacobclark** [comment]
Interested in picking this up as a first bug contribution to Docker, could anybody potentially mentor me through this?

**Comment by cpuguy83** [comment]
@imjacobclark This is most likely going to need to be fixed in github.com/docker/libnetwork.

**Comment by krasi-georgiev** [comment]
@imjacobclark - let me know if you  will work on this as otherwise I will give it a try.

**Comment by krasi-georgiev** [comment]
I found the problem so will submit the pull request soon.

**Comment by hueyvle** [comment]
I'm using docker 19.03.5, and this issue still exist: 

`
 docker info
Client:
 Debug Mode: false

Server:
 Containers: 10
  Running: 8
  Paused: 0
  Stopped: 2
 Images: 14
 Server Version: 19.03.5
 Storage Driver: devicemapper
  Pool Name: docker-252:0-20642-pool
  Pool Blocksize: 65.54kB
  Base Device Size: 10.74GB
  Backing Filesystem: ext4
  Udev Sync Supported: true
  Data file: /dev/loop0
  Metadata file: /dev/loop1
  Data loop file: /opt/docker/lib/296608.296608/devicemapper/devicemapper/data
  Metadata loop file: /opt/docker/lib/296608.296608/devicemapper/devicemapper/metadata
  Data Space Used: 276.9MB
  Data Space Total: 107.4GB
  Data Space Available: 19.48GB
  Metadata Space Used: 18MB
  Metadata Space Total: 2.147GB
  Metadata Space Available: 2.129GB
  Thin Pool Minimum Free Space: 10.74GB
  Deferred Removal Enabled: true
  Deferred Deletion Enabled: true
  Deferred Deleted Device Count: 0
  Library Version: 1.02.137 (2016-11-30)
 Logging Driver: syslog
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: active
  NodeID: fs9icy5ohanizw2r6mxz7d7ta
  Is Manager: true
  ClusterID: vm4dibc0fc5chmspf3pupf7ol
  Managers: 1
  Nodes: 1
  Default Address Pool: 10.0.0.0/8
  SubnetSize: 24
  Data Path Port: 4789
  Orchestration:
   Task History Retention Limit: 5
  Raft:
   Snapshot Interval: 10000
   Number of Old Snapshots to Retain: 0
   Heartbeat Tick: 1
   Election Tick: 10
  Dispatcher:
   Heartbeat Period: 5 seconds
  CA Configuration:
   Expiry Duration: 3 months
   Force Rotate: 0
  Autolock Managers: false
  Root Rotation In Progress: false
  Node Address: 10.10.96.211
  Manager Addresses:
   10.10.96.211:2377
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 894b81a4b802e4eb2a91d1ce216b8817763c29fb
 runc version: 425e105d5a03fabd737a126ad93d62a9eeede87f
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
  userns
 Kernel Version: 4.19.111generic
 Operating System: Debian GNU/Linux 9 (stretch)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 7.775GiB
 Name: esa
 ID: LTB5:OI2M:RCSI:7QWY:E3QB:M75B:B7R5:ZX35:Z3SN:IZNG:ULVG:TPZ7
 Docker Root Dir: /opt/docker/lib/296608.296608
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: the devicemapper storage-driver is deprecated, and will be removed in a future release.
WARNING: devicemapper: usage of loopback devices is strongly discouraged for production use.
         Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.

`
